### PR TITLE
[PWD-2462] Learning Material Page

### PR DESCRIPTION
**Pull request by hiljaker**
### What type of PR is this? (_check all applicable_)

- [x] Feature
- [ ] Bug Fix
- [ ] Documentation Update
- [ ] Setup
- [ ] Code Refactor
- [ ] Performance Improvements
- [ ] Test
- [ ] Build
- [ ] CI
- [ ] Revert

### CODE REVIEW

**Comment by hiljaker** [comment]
‼️ On Progress ‼️
get data dari backend

File: src/components/PWDKChip.tsx [hunk 1]
```diff
@@ -0,0 +1,52 @@
+import { Chip } from '@mui/material'
+import { styled } from '@mui/material/styles'
+import { ComponentProps } from 'react'
+
+interface ChipProps extends ComponentProps<typeof Chip> {
+  label: string
+  selectable?: boolean
+  selected?: boolean
+}
```
Comment (shadiqfauzi) [inline src/components/PWDKChip.tsx:9]:
```suggestion
interface ChipProps extends Chip {
  label: string
  selectable?: boolean
  selected?: boolean
}
```
bukannya gini ya harusnya? mui harusnya sih nyediain types nya chip yang kita bisa extend sesuai kebutuhan

Comment (hiljaker) [reply src/components/PWDKChip.tsx:9]: gabisa a, errornya karena kalo Chip aja dia refers to value, sementara butuhnya type

File: src/views/learning-material/index.tsx
Comment (gangsararyop) [inline src/views/learning-material/index.tsx:1]: buat layout page nya pake container kan ya harusnya? @hiljaker @shadiqfauzi

File: src/views/learning-material/categories/OnlineWorkshop.tsx [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+
+  const [onlineCourses, setOnlineCourses] = useState<ProgramProps[]>([])
+  const [count, setCount] = useState<number>(0)
+  const [limit, setLimit] = useState<number>(initialLimit)
+  const [fetchLoading, setFetchLoading] = useState<boolean>(true)
+  const [loadMoreLoading, setLoadMoreLoading] = useState<boolean>(true)
+
+  const page = useMemo<number>(() => limit / initialLimit, [limit])
+
+  const { data: onlineWorskshopResponse, isLoading: loadingWorkshops } =
```
Comment (kuntiarso) [inline src/views/learning-material/categories/OnlineWorkshop.tsx:33]:
```suggestion
  const { data: onlineWorkshops, isLoading: isLoadingOnlineWorkshops } =
```

File: src/views/learning-material/categories/OnlineWorkshop.tsx [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+import PWDKChip from '~/components/PWDKChip'
+import { useGetOnlineCourses } from '~/features/learning-material/api/useGetOnlineCourses'
+import { useGetOnlineCourseTopics } from '~/features/learning-material/api/useGetOnlineCourseTopics'
+import { OnlineCourseTopic } from '~/features/learning-material/types/OnlineCourseTopic'
+import ProgramProps from '~/features/learning-material/types/ProgramProps'
+import ProgramCardList from '../components/ProgramCardList'
+import useOnlineCourseFormatter from '../hooks/useOnlineCourseFormatter'
+
+const OnlineWorkshop = () => {
+  const { data: topicResponse, isLoading: loadingTopicOptions } =
```
Comment (kuntiarso) [inline src/views/learning-material/categories/OnlineWorkshop.tsx:12]:
```suggestion
  const { data: onlineCourseTopics, isLoading: isLoadingOnlineCourseTopics } =
```

File: src/views/learning-material/categories/SABootcamp.tsx [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+
+  const [onlineCourses, setOnlineCourses] = useState<ProgramProps[]>([])
+  const [count, setCount] = useState<number>(0)
+  const [limit, setLimit] = useState<number>(initialLimit)
+  const [fetchLoading, setFetchLoading] = useState<boolean>(true)
+  const [loadMoreLoading, setLoadMoreLoading] = useState<boolean>(true)
+
+  const page = useMemo<number>(() => limit / initialLimit, [limit])
+
+  const { data: SABResponse, isLoading } = useGetOnlineCourses({
```
Comment (kuntiarso) [inline src/views/learning-material/categories/SABootcamp.tsx:18]:
```suggestion
  const { data: onlineCourseSAB, isLoading } = useGetOnlineCourses({
```

File: src/views/learning-material/categories/VideoLearning.tsx [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+
+  const [onlineCourses, setOnlineCourses] = useState<ProgramProps[]>([])
+  const [count, setCount] = useState<number>(0)
+  const [limit, setLimit] = useState<number>(initialLimit)
+  const [fetchLoading, setFetchLoading] = useState<boolean>(true)
+  const [loadMoreLoading, setLoadMoreLoading] = useState<boolean>(true)
+
+  const page = useMemo<number>(() => limit / initialLimit, [limit])
+
+  const { data: VLResponse, isLoading } = useGetOnlineCourses({
```
Comment (kuntiarso) [inline src/views/learning-material/categories/VideoLearning.tsx:18]:
```suggestion
  const { data: onlineCourseVL, isLoading } = useGetOnlineCourses({
```

File: src/views/learning-material/categories/VideoLearningInEnglish.tsx [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+
+  const [onlineCourses, setOnlineCourses] = useState<ProgramProps[]>([])
+  const [count, setCount] = useState<number>(0)
+  const [limit, setLimit] = useState<number>(initialLimit)
+  const [fetchLoading, setFetchLoading] = useState<boolean>(true)
+  const [loadMoreLoading, setLoadMoreLoading] = useState<boolean>(true)
+
+  const page = useMemo<number>(() => limit / initialLimit, [limit])
+
+  const { data: VLEnglishResponse, isLoading } = useGetOnlineCourses({
```
Comment (kuntiarso) [inline src/views/learning-material/categories/VideoLearningInEnglish.tsx:18]:
```suggestion
  const { data: onlineCourseVLEnglish, isLoading } = useGetOnlineCourses({
```

File: src/views/learning-material/components/ProgramCardList.tsx [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+const ProgramCardList = ({
+  programs,
+  setLimit,
+  fetchLoading,
+  loadMoreLoading,
+  count,
+  limit,
+  initialLimit,
+  ...props
+}: ProgramCardListProps & ComponentProps<typeof Box>) => {
```
Comment (kuntiarso) [inline src/views/learning-material/components/ProgramCardList.tsx:28]:
```suggestion
interface ProgramCardListProps extends ComponentProps<typeof Box> {
  loading?: boolean
  programs?: ProgramProps[]
  setLimit: () => void
  fetchLoading: boolean
  loadMoreLoading: boolean
  count: number
  limit: number
  initialLimit: number
}

const ProgramCardList: FC<ProgramCardListProps> = ({
  programs,
  setLimit,
  fetchLoading,
  loadMoreLoading,
  count,
  limit,
  initialLimit,
  ...props
}) => {
```

File: src/views/learning-material/hooks/useOnlineCourseFormatter.ts [hunk 8]
```diff
@@ -0,0 +1,45 @@
+import { useMemo } from 'react'
+import { OnlineCourseResponse } from '~/features/learning-material/types/OnlineCourse'
+import ProgramProps from '~/features/learning-material/types/ProgramProps'
+
+const useOnlineCourseFormatter = (
+  onlineCourse: OnlineCourseResponse | undefined
+) => {
+  const programs = useMemo(() => {
```
Comment (kuntiarso) [inline src/views/learning-material/hooks/useOnlineCourseFormatter.ts:8]: type return valuenya didefine dulu

File: src/views/learning-material/categories/OnlineWorkshop.tsx (hunk 2 above)
Comment (hiljaker) [reply src/views/learning-material/categories/OnlineWorkshop.tsx:33]: value datanya object, result dan count. nama variablenya gue singularin yaa

**Review by kuntiarso — APPROVED** [review]
gaskeun 🚀

**Review by Vesals — APPROVED** [review]

**Pull request merged**